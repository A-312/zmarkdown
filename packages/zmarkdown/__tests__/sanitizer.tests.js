const remarkConfig = require('../config/remark')
const rebberConfig = require('../config/rebber')

// Disable typographical corrections for easier testing
remarkConfig.noTypography = true
remarkConfig._test = true

const zmarkdown = require('../server')

const renderString = (config = {remarkConfig, rebberConfig}) =>
  (input) =>
    zmarkdown(config).renderString(input).then((vfile) =>
      vfile.toString().trim())

describe('#ping', () => {
  it('do not oversanitize ping', () => {
    const rendered = renderString()('@Clem')
    expect(rendered).resolves.toContain('class="ping ping-link"')
    expect(rendered).resolves.toContain('class="ping-username"')
  })
})

describe('#math', () => {
  it('do not oversanitize math - test sub/sup', () => {
    const rendered = renderString()('$$1_n + 1^n$$')
    expect(rendered).resolves.toMatchSnapshot()
  })

  it('do not oversanitize math - test frac', () => {
    const rendered = renderString()('$$\\frac{1+1}{x+y}$$')
    expect(rendered).resolves.toMatchSnapshot()
  })

  it('do not oversanitize math - test sqrt', () => {
    const rendered = renderString()('$$\\sqrt[3]{x^3 + y^3 + z^3}$$')
    expect(rendered).resolves.toMatchSnapshot()
  })

  it('do not oversanitize math - test notin', () => {
    const rendered = renderString()('$$\\notin$$')
    expect(rendered).resolves.toMatchSnapshot()
  })

  it('do not oversanitize math - test overrightarrow', () => {
    const rendered = renderString()('$$\\overrightarrow{AB}$$')
    expect(rendered).resolves.toMatchSnapshot()
  })
})
