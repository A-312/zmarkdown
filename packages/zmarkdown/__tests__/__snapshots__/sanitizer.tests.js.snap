// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#math do not oversanitize math - test frac 1`] = `"<p><span class=\\"inlineMath inlineMathDouble\\"><span class=\\"katex-display\\"><span class=\\"katex\\"><span class=\\"katex-mathml\\"><math><semantics><mrow><mi>r</mi><mi>a</mi><mi>c</mi><mrow><mn>1</mn><mo>+</mo><mn>1</mn></mrow><mrow><mi>x</mi><mo>+</mo><mi>y</mi></mrow></mrow><annotation>rac{1+1}{x+y}</annotation></semantics></math></span><span class=\\"katex-html\\"><span class=\\"base\\"><span class=\\"strut\\" style=\\"height:0.8388800000000001em;vertical-align:-0.19444em;\\"></span><span class=\\"mord mathdefault\\" style=\\"margin-right:0.02778em;\\">r</span><span class=\\"mord mathdefault\\">a</span><span class=\\"mord mathdefault\\">c</span><span class=\\"mord\\"><span class=\\"mord\\">1</span><span class=\\"mspace\\" style=\\"margin-right:0.2222222222222222em;\\"></span><span class=\\"mbin\\">+</span><span class=\\"mspace\\" style=\\"margin-right:0.2222222222222222em;\\"></span><span class=\\"mord\\">1</span></span><span class=\\"mord\\"><span class=\\"mord mathdefault\\">x</span><span class=\\"mspace\\" style=\\"margin-right:0.2222222222222222em;\\"></span><span class=\\"mbin\\">+</span><span class=\\"mspace\\" style=\\"margin-right:0.2222222222222222em;\\"></span><span class=\\"mord mathdefault\\" style=\\"margin-right:0.03588em;\\">y</span></span></span></span></span></span></span></p>"`;

exports[`#math do not oversanitize math - test sub/sup 1`] = `"<p><span class=\\"inlineMath inlineMathDouble\\"><span class=\\"katex-display\\"><span class=\\"katex\\"><span class=\\"katex-mathml\\"><math><semantics><mrow><msub><mn>1</mn><mi>n</mi></msub><mo>+</mo><msup><mn>1</mn><mi>n</mi></msup></mrow><annotation>1_n + 1^n</annotation></semantics></math></span><span class=\\"katex-html\\"><span class=\\"base\\"><span class=\\"strut\\" style=\\"height:0.79444em;vertical-align:-0.15em;\\"></span><span class=\\"mord\\"><span class=\\"mord\\">1</span><span class=\\"msupsub\\"><span class=\\"vlist-t vlist-t2\\"><span class=\\"vlist-r\\"><span class=\\"vlist\\" style=\\"height:0.151392em;\\"><span style=\\"top:-2.5500000000000003em;margin-left:0em;margin-right:0.05em;\\"><span class=\\"pstrut\\" style=\\"height:2.7em;\\"></span><span class=\\"sizing reset-size6 size3 mtight\\"><span class=\\"mord mathdefault mtight\\">n</span></span></span></span><span class=\\"vlist-s\\">â€‹</span></span><span class=\\"vlist-r\\"><span class=\\"vlist\\" style=\\"height:0.15em;\\"><span></span></span></span></span></span></span><span class=\\"mspace\\" style=\\"margin-right:0.2222222222222222em;\\"></span><span class=\\"mbin\\">+</span><span class=\\"mspace\\" style=\\"margin-right:0.2222222222222222em;\\"></span></span><span class=\\"base\\"><span class=\\"strut\\" style=\\"height:0.7143919999999999em;vertical-align:0em;\\"></span><span class=\\"mord\\"><span class=\\"mord\\">1</span><span class=\\"msupsub\\"><span class=\\"vlist-t\\"><span class=\\"vlist-r\\"><span class=\\"vlist\\" style=\\"height:0.7143919999999999em;\\"><span style=\\"top:-3.113em;margin-right:0.05em;\\"><span class=\\"pstrut\\" style=\\"height:2.7em;\\"></span><span class=\\"sizing reset-size6 size3 mtight\\"><span class=\\"mord mathdefault mtight\\">n</span></span></span></span></span></span></span></span></span></span></span></span></span></p>"`;
